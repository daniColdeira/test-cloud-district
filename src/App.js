import React from 'react';
import Login from './components/GoogleLogin';
import ViewUsers from './components/ViewUsers'; 
import { Route, Switch } from "react-router-dom";
import PrivateRoute from './layout/PrivateRoute'

function App(props) {
  
  return (
    <div>
        <Switch>
          <Route exact path="/" component={Login} />
          <PrivateRoute path="/users">
            <ViewUsers />
          </PrivateRoute>
        </Switch> 
    </div>
  );
}


export default App;
