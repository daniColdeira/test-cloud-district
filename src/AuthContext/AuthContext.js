import React, { createContext, useState } from "react";

export const AuthContext = createContext();

const AppProvider = ({ children }) => {
  const [user, setUser] = useState({user: {}, isAuthenticated: false});
  const setUserAuth = (user, isAuthenticated) => {
    setUser({user, isAuthenticated});
  };
  return (
    <AuthContext.Provider value={{ user, setUserAuth }}>
      {children}
    </AuthContext.Provider>
  );
};

export default AppProvider;