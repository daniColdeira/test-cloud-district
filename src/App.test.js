import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import AuthContext from './AuthContext/AuthContext.js';
import { ConnectedRouter } from 'connected-react-router'
import configureStore, { history } from "./store"; //historial de rutas

const store = configureStore()

test("renders correctly", () => {
    const tree = renderer
    .create(
        <AuthContext>
            <Provider store={store}>
                <ConnectedRouter history={history}>
                    <App />
                </ConnectedRouter>
            </Provider>
        </AuthContext>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

