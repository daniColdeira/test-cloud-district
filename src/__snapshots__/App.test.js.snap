// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="sc-hKFyIo gkzZcg"
  >
    <div
      className="sc-gtssRu jmYPkE"
    >
      <div />
      <div />
    </div>
  </div>
</div>
`;
